wave_optimization/include/wave/optimization/factor_graph/variable.hpp

```diff
@@ -0,0 +1,56 @@
+#ifndef WAVE_OPTIMIZATION_FACTOR_GRAPH_VARIABLE_HPP
+#define WAVE_OPTIMIZATION_FACTOR_GRAPH_VARIABLE_HPP
+
+#include "wave/utils/utils.hpp"
+
+namespace wave {
+
+struct Variable {
+    size_t id;
```

Comment:
http://stackoverflow.com/questions/10747810/what-is-the-difference-between-typedef-and-using-in-c11

See _Effective Modern C++_ Item 9: Prefer alias declarations to typedefs

I prefer `using` even where it gives no extra functionality because
- It looks just like an assignment. `typedef` looks "backwards".
- since `using` is necessary in some cases, it is more consistent to use it everywhere.

>  I think using using is a tad more confusing than using typedef to non-modern C++ developers.

Sure, it is confusing the first time you see it, just like the first time a new C programmer sees `typedef`. (probably less so, because of the backwardness issue).

I'm not opposed to you writing `typedef` if you want.